test-data/unit/fine-grained.test

```diff
@@ -2714,6 +2714,7 @@ class C:
         class D:
             def __init__(self) -> None:
                 self.x = {}
+            def meth(self) -> None:
```

Comment:
What about also testing that the type inference works correctly in fine-grained mode (within one method)?